enumerable_methods.rb

```diff
@@ -0,0 +1,156 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    ind = 0
+    for i in self
+      yield(i, ind)
+      ind += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << n if yield(n)
+    end
+    new_arr
+  end
+
+  def my_all?
+    # return true if empty array is passed, regardless of block_given
+    return true if (self.class == Array && count.zero?) || (!count.zero? && !block_given?)
+
+    my_each do |n|
+      return false unless yield(n)
+    end
+    true
+  end
+
+  def my_any?
+    return false if self.class == Array && count.zero?
+    return true if !count.zero? && !block_given?
+
+    my_each do |n|
+      return true if yield(n)
+    end
+    false
+  end
+
+  def my_none?
+    return true if self.class == Array && count.zero?
+    return false if !count.zero? && !block_given?
+
+    my_each do |n|
+      return false if yield(n)
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    if !arg.nil?
+      my_each do |n|
+        counter += 1 if n == arg
+      end
+    elsif block_given?
+      my_each do |n|
+        counter += 1 if yield(n)
+      end
+    else
+      counter = self.length
+    end
+    counter
+  end
+
+=begin
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << yield(n)
+    end
+    new_arr
+  end
+=end
+
+  # my_map modified to accept either block or proc (i.e., a named block)
+
+  def my_map(&block)
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << block.call(n)
+    end
+    new_arr
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```